pkg/topo/get.go

```diff
@@ -0,0 +1,225 @@
+}
+
+func listObjects(cmd *cobra.Command) ([]topoapi.Object, error) {
+	conn, err := cli.GetConnection(cmd)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	client := topoapi.CreateTopoClient(conn)
+
+	resp, err := client.List(context.Background(), &topoapi.ListRequest{})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Objects, nil
+}
+
+func getObject(cmd *cobra.Command, id topoapi.ID) (*topoapi.Object, error) {
+	conn, err := cli.GetConnection(cmd)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	client := topoapi.CreateTopoClient(conn)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	response, err := client.Get(ctx, &topoapi.GetRequest{ID: id})
+	if err != nil {
+		cli.Output("get error")
+		return nil, err
+	}
+	return response.Object, nil
+}
+
+func printHeader(writer io.Writer, objectType topoapi.Object_Type, verbose bool, printUpdateType bool) {
+	if printUpdateType {
+		_, _ = fmt.Fprintf(writer, "%-*.*s", width, prec, "Update Type")
+	}
+
+	if objectType == topoapi.Object_ENTITY {
+		_, _ = fmt.Fprintf(writer, "%-*.*s%-*.*s\n%-*.*s", width, prec, "Object Type",
+			width, prec, "Entity ID", width, prec, "Kind ID")
+	} else if objectType == topoapi.Object_RELATION {
+		_, _ = fmt.Fprintf(writer, "%-*.*s%-*.*s\n%-*.*s%-*.*s%-*.*s", width, prec, "Object Type",
+			width, prec, "Relation ID", width, prec, "Kind ID", width, prec, "Source ID", width, prec, "Target ID")
+	} else if objectType == topoapi.Object_KIND {
+		_, _ = fmt.Fprintf(writer, "%-*.*s%-*.*s\n%-*.*s", width, prec, "Object Type",
+			width, prec, "Kind ID", width, prec, "Name")
+	}
+
+	if !verbose {
+		_, _ = fmt.Fprintf(writer, "\tAspects\n")
+	}
+}
+
+const (
+	width = 16
+	prec  = width - 1
+)
+
+func printObject(writer io.Writer, object topoapi.Object, verbose bool) {
+	switch object.Type {
+	case topoapi.Object_ENTITY:
+		var kindID topoapi.ID
+		if e := object.GetEntity(); e != nil {
+			kindID = e.KindID
+		}
+		_, _ = fmt.Fprintf(writer, "%-*.*s%-*.*s%-*.*s", width, prec, object.Type, width, prec, object.ID, width, prec, kindID)
+		printAspects(writer, object, verbose)
+
+	case topoapi.Object_RELATION:
+		r := object.GetRelation()
+		_, _ = fmt.Fprintf(writer, "%-*.*s%-*.*s%-*.*s%-*.*s%-*.*s", width, prec, object.Type, width, prec, object.ID, width, prec, r.KindID,
+			width, prec, r.SrcEntityID, width, prec, r.TgtEntityID)
+		printAspects(writer, object, verbose)
+
+	case topoapi.Object_KIND:
+		k := object.GetKind()
+		_, _ = fmt.Fprintf(writer, "%-*.*s%-*.*s%-*.*s", width, prec, object.Type, width, prec, object.ID, width, prec, k.GetName())
+		printAspects(writer, object, verbose)
+
+	default:
+		_, _ = fmt.Fprintf(writer, "\n")
+	}
+}
+
+func printAspects(writer io.Writer, object topoapi.Object, verbose bool) {
+	if verbose {
+		for aspectType, aspect := range object.Aspects {
+			_, _ = fmt.Fprintf(writer, "\t%s=%s\n", aspectType, bytes.NewBuffer(aspect.Value).String())
```

Comment:
Is this will be printed in JSON format?

Reply:
I mean when we have more complex aspects to print, we will see them in a well formed JSON?

Reply:
Yes.